index.js

```diff
@@ -71,5 +73,67 @@ export default function icomesh(order = 4) {
         vertices[i + 2] *= m;
     }
 
-    return {vertices, triangles};
+    if (!uvMap) return {vertices, triangles};
+
+    // uv mapping
+    const uv = new Float32Array((numVertices + numDuplicates) * 2);
+    for (let i = 0; i < numVertices; i++) {
+        uv[2 * i + 0] = Math.atan2(vertices[3 * i + 2], vertices[3 * i]) / (2 * Math.PI) + 0.5;
+        uv[2 * i + 1] = Math.asin(vertices[3 * i + 1]) / Math.PI + 0.5;
+    }
+
+    const duplicates = new Map();
+
+    function addDuplicate(i, uvx, uvy, cached) {
+        if (cached) {
+            const dupe = duplicates.get(i);
+            if (dupe !== undefined) return dupe;
+        }
+        vertices[3 * v + 0] = vertices[3 * i + 0];
+        vertices[3 * v + 1] = vertices[3 * i + 1];
+        vertices[3 * v + 2] = vertices[3 * i + 2];
+        uv[2 * v + 0] = uvx;
+        uv[2 * v + 1] = uvy;
+        if (cached) duplicates.set(i, v);
+        return v++;
+    }
+
+    for (let i = 0; i < triangles.length; i += 3) {
```

Comment:
You could cache `triangles.length` to `len = triangles.length`. It's better for small set / non-hot patched cases.